From c4dd27cc69b99e13d4117fa348f968cf2199db6b Mon Sep 17 00:00:00 2001
From: Philipp Reisner <philipp.reisner@linbit.com>
Date: Wed, 28 Aug 2024 18:43:31 +0200
Subject: [PATCH 07/32] build: Another update to the spdx files

Some enterprise customers require these files to contain a
"PackageCopyrightText" tag. Apart from that they require the
SPDX in JSON format. That will come in another commit.
---
 Makefile | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3f58568cc138..80c20d24ab54 100644
--- a/Makefile
+++ b/Makefile
@@ -205,6 +205,8 @@ PackageSupplier: Organization: LINBIT HA-Solutions GmbH
 PackageDownloadLocation: https://github.com/LINBIT/drbd
 FilesAnalyzed: false
 PackageLicenseDeclared: GPL-2.0-only
+PackageCopyrightText: <text>2001-2008, LINBIT Information Technologies GmbH
+2008-$(SPDX_YEAR), LINBIT HA-Solutions GmbH</text>
 Relationship: SPDXRef-DOCUMENT DESCRIBES SPDXRef-Package-$(SPDX_PKG_NAME)
 endef
 
@@ -219,6 +221,7 @@ drbd-kmod_rhel.spdx drbd-kmod_sles.spdx:
 		SPDX_DATE="$$(date --utc +%FT%TZ)"; \
 		SPDX_UUID="$$(cat /proc/sys/kernel/random/uuid)"; \
 		SPDX_VERSION="$(REL_VERSION)"; \
+		SPDX_YEAR="$$(date --utc +%Y)"; \
 		case "$@" in \
 			drbd-kmod_rhel.spdx) SPDX_PKG_NAME=kmod-drbd;; \
 			drbd-kmod_sles.spdx) SPDX_PKG_NAME=drbd-kmp-default;; \
@@ -228,11 +231,13 @@ drbd-kmod_rhel.spdx drbd-kmod_sles.spdx:
 		test -n "$$SPDX_DATE"; \
 		test -n "$$SPDX_UUID"; \
 		test -n "$$SPDX_VERSION"; \
+		test -n "$$SPDX_YEAR"; \
 		$(MAKE) spdx-file SPDX_UUID="$$SPDX_UUID" \
 			SPDX_DATE="$$SPDX_DATE" \
 			SPDX_FILE_TMP="$@.tmp" \
 			SPDX_PKG_NAME="$$SPDX_PKG_NAME" \
-			SPDX_VERSION="$$SPDX_VERSION"; \
+			SPDX_VERSION="$$SPDX_VERSION" \
+			SPDX_YEAR="$$SPDX_YEAR"; \
 		mv $@.tmp $@; )
 
 # only call this wrapper from drbd-kmod.cdx.json
-- 
2.35.3

